src/lib/autoMode.js

```diff
@@ -0,0 +1,86 @@
+import appState from './appState';
+
+let delayTime, incoming, scene, presets;
+
+export function initAutoMode(_scene) {
+  scene = _scene;
+
+  const auto = appState.getQS().get('auto');
+  if (!auto) {
+    return;
+  }
+
+  if (!/s$/i.test(auto)) {
+    console.error('malformed auto param; missing seconds specifier');
+    return;
+  }
+
+  let seconds = parseFloat(auto.replace(/s$/i, ''));
+  if (Number.isNaN(seconds)) {
+    console.error('malformed auto param; not a number');
+    return;
+  }
+
+  if (seconds <= 1) {
+    console.warn('auto param seconds too small; defaulting to 30');
+    seconds = 30;
+  }
+
+  delayTime = seconds * 1000;
+
+  getJSON('/static/autoPresets.json', function(data) {
```

Comment:
@anvaka I'm having trouble here... `http://localhost:8880/static/autoPresets.json` doesn't seem to exist. Any tips on getting the server to actually serve the static files?

Reply:
You can use standard webpack's `require.ensure()` for this. Let's say `autoPresets.json` are in the `src/lib` folder. Then you can do:

``` js
require.ensure('@/lib/autoPresets.json', () => {
  const presets = require('@/lib/autoPresets.json');
  // ...
});
```

This will split files correctly. But to be honest, I think this is somewhat premature at the moment. I'd just have synchronous version:

``` js
const presets = require('@/lib/autoPresets.json');
 ...
```

Or maybe even better version, is to make `autoPresets` a javascript module, then we'd be able to augment it with dynamic features, and nicer javascript syntax for the field:

``` js
// autoPresets.js
module.exports = [{
  "name": "Combination of two fields",
  "dt": 0.01,
  "fo": 0.998,
  "dp": 0.009,
  "cm": 1,
  "cx": 0.000300000000000189,
  "cy": 0,
  "w": 8.5404,
  "h": 8.5404,
  "vf": `// p.x and p.y are current coordinates
  // v.x and v.y is a velocity at point p
  vec2 get_velocity(vec2 p) {
  vec2 v = vec2(0., 0.);

  // change this to get a new vector field
  vec2 p1 = p - vec2(-2., 0.);
  vec2 p2 = p - cursor.zw;

  float l1 = length(p1), l2 = length(p2);

  v = vec2(-p1.y, p1.x)/(l1 * l1) + vec2(-p2.y, p2.x)/(l2 * l2);

  return v;
}`
},
...
]
```

This would also let us drop `util/autoPresets.js`